### PR TITLE
Create default incoming based on opts.host and opts.port

### PR DESCRIPTION
**Pull request by regular**
Fixes a bug that ignores host and port given in the config file.
Variable "overrides" only includes command-line overrides, not the contents of the
config file. Therfore the default incoming connection alsways uses the
default port, not the port given in the config file.

(This also removes dupicate key "party")

### CODE REVIEW

**Comment by regular** [comment]
@christianbundy yes, it does. What is hardcoded is the _default_ port, the cli overrides are merge in. This is no different than with all the other settings. The code I removed was a failing attempt to make a default incoming connection that would respect user preferences for host and port. Instead it did _only_ respect _command line_ overrides and did ignore the user's config file.

It now creates the default incoming connection _after_ RC has _combined_ defaults, config file settings and cli overrides, so all of them are respected when creating the incoming connection.

This fixes a bug where `{port: xxx, host: xxx}` in .`ssb/config` simply is ignored, which breaks client applications that don't use the default port.

There is now an integration-level test that will hopefully catch this in the future: https://github.com/regular/ssb-integration-tests

**Comment by regular** [comment]
It would be good if we can move fast on this, and also to get this into scuttlebot-release, because until we do, sbot is broken, and sbot is critical infrastructure. cc @dominictarr @arj03 @cryptix

File: inject.js [hunk 1]
```diff
@@ -12,15 +12,14 @@ var MIN = 60*SEC
...
-  return RC(name || 'ssb', merge({
+  var result = RC(name || 'ssb', merge({
     //just use an ipv4 address by default.
     //there have been some reports of seemingly non-private
     //ipv6 addresses being returned and not working.
     //https://github.com/ssbc/scuttlebot/pull/102
     party: true,
     host: nonPrivate.v4 || '',
-    port: port,
+    port: 8008,
```
Comment (dominictarr) [inline inject.js]: why did you remove `var port = override && override.port || 8008` ? oh hang on, it can be removed because the override data is merged into this anyway

**Comment by dominictarr** [comment]
@regular awesome! merged into 2.3.5

**Comment by dominictarr** [comment]
will be in scuttlebot-release@13.0.0

**Comment by dominictarr** [comment]
@regular if you do `sbot.getAddress` it will give you an multiaddress containing a ws address.
Ideally, you should just use that straight with `ssb-client` in the browser and it will just ignore the parts it doesn't understand. (there may be some bugs to fix about that though)

**Comment by regular** [comment]
@dominictarr we could easily keep ws.getAddress around to not break the client API though.

See also 
- `%VVu0cOcIyoalcJUxAtqHu0BXkh8rxdBZuZIdlS5MmUQ=.sha256`
- `%X/jJMG+nepTB58Prfs98H25MqEO8c9yaamBzgikZn1I=.sha256`

**Comment by regular** [comment]
@dominictarr in scuttlebot@13 `getAddress` returns one address only, a `net` address *and* ssb-client in the browser even tries to use it, resulting in `net.connect is not a function`.  Now every browser client relying on the stock sbot binary is completely broken. :(
